contracts/KyberPayWrapper.sol

```diff
@@ -0,0 +1,181 @@
+            dest,
+            destAddress,
+            maxDestAmount,
+            minConversionRate,
+            walletId,
+            paymentData,
+            hint,
+            kyberNetworkProxy
+        );
+
+        uint paidAmount = (src == dest) ? doPayWithoutKyber(payData) : doPayWithKyber(payData);
+
+        // log as event
+        ProofOfPayment(destAddress, dest, paidAmount, paymentData);
+    }
+
+    function doPayWithoutKyber(PayData memory payData) internal returns (uint paidAmount) {
+
+        uint returnAmount;
+
+        if (payData.srcAmount > payData.maxDestAmount) {
+            paidAmount = payData.maxDestAmount;
+            returnAmount = payData.srcAmount - payData.maxDestAmount;
+        } else {
+            paidAmount = payData.srcAmount;
+            returnAmount = 0;
+        }
+
+        if (payData.src == ETH_TOKEN_ADDRESS) {
+            payData.destAddress.transfer(paidAmount);
+
+            // return change
+            if (returnAmount > 0) msg.sender.transfer(returnAmount);
+        } else {
+            require(payData.src.transferFrom(msg.sender, payData.destAddress, paidAmount));
+        }
+    }
+
+    function doPayWithKyber(PayData memory payData) internal returns (uint paidAmount) {
+
+        uint returnAmount;
+        uint wrapperSrcBalanceBefore;
+        uint destAddressBalanceBefore;
+        uint wrapperSrcBalanceAfter;
+        uint destAddressBalanceAfter;
+        uint srcAmountUsed;
+
+        if (payData.src != ETH_TOKEN_ADDRESS) {
+            require(payData.src.transferFrom(msg.sender, address(this), payData.srcAmount));
+            require(payData.src.approve(payData.kyberNetworkProxy, 0));
+            require(payData.src.approve(payData.kyberNetworkProxy, payData.srcAmount));
+        }
+
+
+        (wrapperSrcBalanceBefore, destAddressBalanceBefore) = getBalances(
+            payData.src,
+            payData.dest,
+            payData.destAddress
+        );
+
+        paidAmount = doTradeWithHint(payData);
+        if (payData.src != ETH_TOKEN_ADDRESS) require(payData.src.approve(payData.kyberNetworkProxy, 0));
+
+        (wrapperSrcBalanceAfter, destAddressBalanceAfter) = getBalances(payData.src, payData.dest, payData.destAddress);
+
+        // verify the amount the user got is same as returned from Kyber Network
+        require(destAddressBalanceAfter > destAddressBalanceBefore);
+        require(paidAmount == (destAddressBalanceAfter - destAddressBalanceBefore));
+
+        // calculate the returned change amount
+        require(wrapperSrcBalanceBefore >= wrapperSrcBalanceAfter);
+        srcAmountUsed = wrapperSrcBalanceBefore - wrapperSrcBalanceAfter;
+
+        require(payData.srcAmount >= srcAmountUsed);
+        returnAmount = payData.srcAmount - srcAmountUsed;
+
+        // return to sender the returned change
+        if (returnAmount > 0) {
+            if (payData.src == ETH_TOKEN_ADDRESS) msg.sender.transfer(returnAmount);
+            else {
```

Comment:
why not in one line?

Reply:
as we saw it causes linter problems.
changed it to use curly braces.